pkg/cloudprovider/gce/gce.go

```diff
@@ -421,6 +464,57 @@ func (gce *GCECloud) GetNodeResources(name string) (*api.NodeResources, error) {
 	}
 }
 
+func getMetadataValue(metadata *compute.Metadata, key string) string {
+	for _, item := range metadata.Items {
+		if item.Key == key {
+			return item.Value
+		}
+	}
+	return ""
+}
+
+func (gce *GCECloud) Configure(name string, spec *api.NodeSpec) error {
+	instanceName := canonicalizeInstanceName(name)
+	instanceCall := gce.service.Instances.Get(gce.projectID, gce.zone, instanceName)
```

Comment:
Take a look at the getMetadata function above. Each instance has a local (in the host OS) metadata server that it can curl directly. This avoids the network latency, and reduces quota usage for the GCE API.